### ISSUE TITLE
UnhandledException IAsyncResult

### ISSUE THREAD

**Issue by kenetik** [MediaBrowser/Emby#1987]
Since upgrading to Emby Server Version 3.0.6000, I am experiencing playback issues almost constantly.

When I attempt to skip forward, or begin a video from a "Scene" (point), 75-80% of the time, the interface becomes non-responsive and hangs with out error. 

Reviewing the server logs, however, provides the following error:
```
UnhandledException
Either the IAsyncResult object did not come from the corresponding async method on this type, or the End method was called multiple times with the same IAsyncResult.
Parameter name: asyncResult
System.ArgumentException
  at System.Threading.Tasks.TaskFactory`1+FromAsyncTrimPromise`1[TResult,TInstance].CompleteFromAsyncResult (IAsyncResult asyncResult) <0x7fc4e1f9a490 + 0x00227> 24799 in <filename unknown>:0
  at System.Net.Sockets.SocketAsyncResult.<ExecuteWorkItem>m__1 (System.Object state) <0x40e422f0 + 0x0009f> in <filename unknown>:0
  at (wrapper managed-to-native) System.Runtime.Remoting.Messaging.AsyncResult:Invoke (System.Runtime.Remoting.Messaging.AsyncResult)
  at System.Runtime.Remoting.Messaging.AsyncResult.System.Threading.IThreadPoolWorkItem.ExecuteWorkItem () <0x7fc4e1e89e30 + 0x0000c> in <filename unknown>:0
  at System.Threading.ThreadPoolWorkQueue.Dispatch () <0x7fc4e1d72e90 + 0x001d6> in <filename unknown>:0
  at System.Threading._ThreadPoolWaitCallback.PerformWaitCallback () <0x7fc4e1d74710 + 0x00008> in <filename unknown>:0
```

I've only started experiencing this issue since upgrading, however it makes usability virtually impossible for my needs. 

```
# uname -a
Linux redacted.hostname 4.4.0-28-generic #47-Ubuntu SMP Fri Jun 24 10:09:13 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux
```

I have a "Pro/Supporter" license, if that's worth anything.

**Comment by LukePulverenti** [comment]
Thanks. Please attach complete log files, thanks.

**Comment by kenetik** [comment]
@LukePulverenti Thanks for the immediate response!

Can you please advise which log file(s) you would like the full output from? I'll be more than happy to provide them.

**Comment by LukePulverenti** [comment]
Any log file that was modified during the time frame.

**Comment by kenetik** [comment]
Here is the full output from `/var/log/emby-server.log` from server startup to crash during this issue:

[Gist: emby-server.log](https://gist.github.com/kenetik/1fd6c6644e46add02d04609c87dd538c)

[emby-server_log.txt](https://github.com/MediaBrowser/Emby/files/386651/emby-server_log.txt)

**Comment by vdbxavier** [comment]
i have the same issue (debian 8.5)

UnhandledException
Either the IAsyncResult object did not come from the corresponding async method on this type, or the End method was called multiple times with the same IAsyncResult.
Parameter name: asyncResult
System.ArgumentException
  at System.Threading.Tasks.TaskFactory`1+FromAsyncTrimPromise`1[TResult,TInstance].CompleteFromAsyncResult (IAsyncResult asyncResult) <0x7f38935fd1f0 + 0x00227> 24797 in <filename unknown>:0
  at System.Net.Sockets.SocketAsyncResult.<ExecuteWorkItem>m__1 (System.Object state) <0x415f43b0 + 0x0009f> in <filename unknown>:0
  at (wrapper managed-to-native) System.Runtime.Remoting.Messaging.AsyncResult:Invoke (System.Runtime.Remoting.Messaging.AsyncResult)
  at System.Runtime.Remoting.Messaging.AsyncResult.System.Threading.IThreadPoolWorkItem.ExecuteWorkItem () <0x7f38934ecc50 + 0x0000c> in <filename unknown>:0
  at System.Threading.ThreadPoolWorkQueue.Dispatch () <0x7f38933d5cc0 + 0x001d6> in <filename unknown>:0
  at System.Threading._ThreadPoolWaitCallback.PerformWaitCallback () <0x7f38933d7530 + 0x00008> in <filename unknown>:0

**Comment by LukePulverenti** [comment]
I would try 3.0.6020, along with the newly released mono 4.4.2. If there's still an issue then you might want to drop back to mono 4.2.3.4.

**Comment by vdbxavier** [comment]
i don't have mono 4.4.2  ...

root@mediacenter:~# ps ax |grep mono
13214 ?        Ss     0:00 su -s /bin/sh -c exec "$0" "$@" www-data -- env MONO_THREADS_PER_CPU=250 MONO_GC_PARAMS=nursery-size=128m /usr/bin/mono-sgen --optimize=all /usr/lib/emby-server/bin/MediaBrowser.Server.Mono.exe -programdata /var/lib/emby-server -restartpath /usr/lib/emby-server/restart.sh
13248 ?        Ssl    0:07 /usr/bin/mono-sgen --optimize=all /usr/lib/emby-server/bin/MediaBrowser.Server.Mono.exe -programdata /var/lib/emby-server -restartpath /usr/lib/emby-server/restart.sh
13292 pts/0    S+     0:00 grep mono


root@mediacenter:~# /usr/bin/mono-sgen --version
Mono JIT compiler version 4.2.1 (Stable 4.2.1.102/6dd2d0d Thu Dec  3 03:58:58 UTC 2015)
Copyright (C) 2002-2014 Novell, Inc, Xamarin Inc and Contributors. www.mono-project.com
	TLS:           __thread
	SIGSEGV:       altstack
	Notifications: epoll
	Architecture:  amd64
	Disabled:      none
	Misc:          softdebug
	LLVM:          supported, not enabled.
	GC:            sgen

emby version : Version 3.0.6060.0

debian stable
